activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -916,6 +920,13 @@ def set_value(name, value) # :nodoc:
 
     private
 
+      # Extract column names from arguments passed to #order or #reorder.
+      def column_names_from_order_arguments(args)
+        args = args.dup
+        orderings = args.extract_options!
```

Comment:
Yeah, that won't work because the hash keys aren't all symbols.

I do think this shouldn't be using `extract_options!` -- I think it wants something more like `args.flat_map { |a| a.is_a?(Hash) ? a.keys : a }`